### PR TITLE
feat(bd-integration-to-auth): added bd-check before auth

### PR DESCRIPTION
**Pull request by DaniWitcher**
added bd-check before auth

### CODE REVIEW

File: projects/core/src/lib/authorization/components/role-select/role-select.component.ts [hunk 1]
```diff
@@ -59,28 +78,33 @@ export class RoleSelectComponent {
...
-		//toDo : Need page to add route in future
+		this.userService.saveUser(this.getUser(Roles.EXPERT)).subscribe();
+		//toDO redirect to Expert page
 	}
 
 	private roleRM(): void {
-		this.getUser(Roles.RM);
-		// toDo : Need page to add route in future
+		this.userService.saveUser(this.getUser(Roles.RM)).subscribe();
+		//toDo redirect to RM page
```
Comment (Tsurhanau) [inline projects/core/src/lib/authorization/components/role-select/role-select.component.ts:92]: plrase add .pipe(takeUntil(this.destroy$))

File: projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts [hunk 2]
```diff
@@ -36,15 +42,39 @@ export class SignInComponent implements OnDestroy {
...
-				//toDo DB service user check
-				// toDo: Integrate DB Service
-			} else {
-				this.router.navigate([Routes.ROLE_SELECT]);
-			}
+			this.userCheck(user.uid!);
 		}
 	}
 
+	private userCheck(uid: String): void {
```
Comment (Tsurhanau) [inline projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts:49]: **s**tring

File: projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts [hunk 3]
```diff
@@ -36,15 +42,39 @@ export class SignInComponent implements OnDestroy {
...
+		this.userServise
+			.getUser()
+			.pipe(takeUntil(this.destroy$))
+			.subscribe((data) => {
+				let isUser = false;
+				let userData!: User;
+				if (data) {
+					for (let user of Object.entries(data)) {
```
Comment (Tsurhanau) [inline projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts:57]: es6

File: projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts [hunk 4]
```diff
@@ -36,15 +42,39 @@ export class SignInComponent implements OnDestroy {
...
-			} else {
-				this.router.navigate([Routes.ROLE_SELECT]);
-			}
+			this.userCheck(user.uid!);
 		}
 	}
 
+	private userCheck(uid: String): void {
+		this.userServise
+			.getUser()
```
Comment (Tsurhanau) [inline projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts:51]: getUserById

File: projects/core/src/lib/authorization/components/role-select/role-select.component.ts (hunk 1 above)
Comment (DaniWitcher) [reply projects/core/src/lib/authorization/components/role-select/role-select.component.ts:92]: Done

File: projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts (hunk 2 above)
Comment (DaniWitcher) [reply projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts:49]: Done

File: projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts (hunk 3 above)
Comment (DaniWitcher) [reply projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts:57]: Done

File: projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts (hunk 4 above)
Comment (DaniWitcher) [reply projects/core/src/lib/authorization/components/sign-in/sign-in.component.ts:51]: Done

**Pull request merged**